setup.py

```diff
@@ -1,5 +1,63 @@
-from setuptools import setup
+from __future__ import division, absolute_import, print_function
+
 import os
+from os import path
+import sys
+from setuptools.dist import Distribution
+from setuptools import setup, Command
+
+
+class CustomDistribution(Distribution):
+    def __init__(self, *args, **kwargs):
+        self.sdist_requires = None
+        Distribution.__init__(self, *args, **kwargs)
+
+    def get_finalized_command(self, command, create=1):
+        cmd_obj = self.get_command_obj(command, create)
+        cmd_obj.ensure_finalized()
+        return cmd_obj
+
+    def export_live_eggs(self, env=False):
+        """Adds all of the eggs in the current environment to PYTHONPATH."""
+        path_eggs = [p for p in sys.path if p.endswith('.egg')]
+
+        command = self.get_finalized_command("egg_info")
+        egg_base = path.abspath(command.egg_base)
+
+        unique_path_eggs = set(path_eggs + [egg_base])
+
+        os.environ['PYTHONPATH'] = ':'.join(unique_path_eggs)
+
+
+class test(Command):  # noqa: ignore=N801
+    """Command to run tox."""
+
+    description = "run tox tests"
+
+    user_options = [('tox-args=', 'a', "Arguments to pass to tox")]
+
+    def initialize_options(self):
+        self.tox_args = ''
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        # Install test dependencies if needed.
+        if self.distribution.tests_require:
+            self.distribution.fetch_build_eggs(self.distribution.tests_require)
+
+        # Add eggs to PYTHONPATH. We need to do this to ensure our eggs are
+        # seen by Tox.
+        self.distribution.export_live_eggs()
```

Comment:
Well the primary issue is that Tox depends on VirtualEnv (which is installed by `fetch_build_eggs`), so we need to add that to `PYTHONPATH` so that Tox sees it.